### PR TITLE
[chore] refactor otlpreceiver start logic, add lots of tests

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: receiver/otlpreceiver/otlp.go [hunk 1]
```diff
@@ -79,13 +76,35 @@ func newOtlpReceiver(cfg *Config, set *receiver.CreateSettings) (*otlpReceiver,
 	return r, nil
 }
 
-func (r *otlpReceiver) startGRPCServer(cfg *configgrpc.GRPCServerSettings, host component.Host) error {
-	r.settings.Logger.Info("Starting GRPC server", zap.String("endpoint", cfg.NetAddr.Endpoint))
+func (r *otlpReceiver) startGRPCServer(host component.Host) error {
+	// If GRPC is not enable, nothing to start.
```
Comment (atoulme) [inline receiver/otlpreceiver/otlp.go:80]:
```suggestion
	// If GRPC is not enabled, nothing to start.
```

File: receiver/otlpreceiver/otlp.go [hunk 2]
```diff
@@ -97,13 +116,46 @@ func (r *otlpReceiver) startGRPCServer(cfg *configgrpc.GRPCServerSettings, host
 	return nil
 }
 
-func (r *otlpReceiver) startHTTPServer(cfg *confighttp.HTTPServerSettings, host component.Host) error {
-	r.settings.Logger.Info("Starting HTTP server", zap.String("endpoint", cfg.Endpoint))
+func (r *otlpReceiver) startHTTPServer(host component.Host) error {
+	// If HTTP is not enable, nothing to start.
```
Comment (atoulme) [inline receiver/otlpreceiver/otlp.go:120]:
```suggestion
	// If HTTP is not enabled, nothing to start.
```

File: receiver/otlpreceiver/otlp.go [hunk 3]
```diff
@@ -97,13 +116,46 @@ func (r *otlpReceiver) startGRPCServer(cfg *configgrpc.GRPCServerSettings, host
...
+			handleMetrics(resp, req, httpMetricsReceiver)
+		})
+	}
+
+	if r.nextLogs != nil {
+		httpLogsReceiver := logs.New(r.nextLogs, r.obsrepHTTP)
+		httpMux.HandleFunc(r.cfg.HTTP.LogsURLPath, func(resp http.ResponseWriter, req *http.Request) {
+			handleLogs(resp, req, httpLogsReceiver)
+		})
+
```
Comment (atoulme) [inline receiver/otlpreceiver/otlp.go:145]:
```suggestion
```

Comment (bogdandrutu) [reply receiver/otlpreceiver/otlp.go]: Done.

File: receiver/otlpreceiver/otlp.go (hunk 2 above)
Comment (bogdandrutu) [reply receiver/otlpreceiver/otlp.go]: Done.

File: receiver/otlpreceiver/otlp.go (hunk 1 above)
Comment (bogdandrutu) [reply receiver/otlpreceiver/otlp.go]: Done.

**Review by atoulme — APPROVED** [review]

**Pull request merged**